### ISSUE TITLE
VS 2017 csproj is incompatible with pre-2017 Unity

### ISSUE THREAD

**Issue by saki7** [pardeike/Harmony#87]
I'm writing a mod for Cities: Skylines (the game's built on pre-2017 UnityEngine), and seeing same errors as described in NuGet/Home#5460. Build fails due to NuGet errors, and there's no way of removing NuGet completely.

Harmony's csproj has been updated to the new VS 2017 project format in #73. It was working fine with older version of csproj. I also confirmed that the old csproj is still working in my sln. Are we going to drop this old Unity support, or is there any workaround for this?

**Comment by pardeike** [comment]
Not sure what you mean. I develop for RimWorld which uses Unity 5.6.3 and I build for target=3.5

**Comment by saki7** [comment]
When I add Harmony csproj to my game plugin's sln, I get errors saying that NuGet is incompatible for `Unity 3.5 full` (note that this is not the same as version 3.5 of .NET Framework)

**Comment by pardeike** [comment]
What’s wrong with building a binary release and embedding it in your project?

**Comment by saki7** [comment]
Few reasons:

- I need to lock my solution's Harmony dependency to a specific version
- I need Harmony as a Project Reference in order to have intellisense completion
- It feels not right to bundle an arbitrary binary dependency when you distribute a plugin for a proprietary game

**Comment by saki7** [comment]
@pardeike Yeah, I usually use official binary distribution for such libraries. But the Unpatch functinality is not included in latest release, no?

**Comment by pardeike** [comment]
No. Not yet. The reason being that Harmony grew out of the RimWorld community and thus has a quite large legacy to support. I recently moved on and fixed a few shortcomings with important things such as try-catch support in the underlying mechanics that required a way to solve the shared API conflicts that occur since Harmony is sharing state between different mods in the same host application.

So in order to prevent lots of unaware users in RimWorld to use the new release and breaking every other mod doing so, I had(have) to delay the next release more than I would like. 

Two weeks ago, I found a workaround by having Harmony patch itself and started to use it actively in one of my more popular mods (20k users on steam) and it was a rocky journey. I am now on a seemingly stable situation and all that is left is to clean up and test the master branch. Once that is done I can get back on short incremental updates.
